airflow/executors/kubernetes_executor.py

```diff
@@ -589,6 +598,44 @@ def sync(self) -> None:
                 break
         # pylint: enable=too-many-nested-blocks
 
+        # Run any pending timed events
+        next_event = self.event_scheduler.run(blocking=False)
+        self.log.debug("Next timed event is in %f", next_event)
+
+    def _check_worker_pods_pending_timeout(self):
+        """Check if any pending worker pods have timed out"""
+        timeout = self.kube_config.worker_pods_pending_timeout
+        self.log.debug('Looking for pending worker pods older than %d seconds', timeout)
+
+        kwargs = {
+            'limit': self.kube_config.worker_pods_pending_timeout_batch_size,
+            'field_selector': 'status.phase=Pending',
+            'label_selector': f'airflow-worker={self.scheduler_job_id}',
```

Comment:
I've limited it only pending pods from this scheduler, but it could easily be expanded to do any with `kubernetes_executor=True` instead (though, seems this could be a race between schedulers).

Reply:
Yeah I think it would be better to do `kubernetes_executor=True` and catch error (`NotFound` and similar errors) in that loop to avoid race condition.

Reply:
The only downside I can see with that is if you had more than 1 instance of Airflow running in a namespace, or on the cluster with multi-namespace mode, you could timeout other instances tasks. Assuming they all have the same timeout it's fine 🤠. Even scheduler_job_id is brittle, just less so. Do we have a better 'unique' something per instance we could use?

Reply:
@kaxil, after sitting down to switch to `kubernetes_executor=True` a third time now, I'm leaning towards leaving it based on `scheduler_job_id` for now. I think it is closer to "right", but what we really need is a unique id for the airflow instance itself (most of KubernetsExecutor would actually benefit from having one, in fact).